fix(core): fix policy synthesizer logic for precreated roles (#31710)
aws/aws-cdk: packages/aws-cdk-lib/aws-iam/README.md

For more information on configuring permissions see the [Security And Safety Dev
Guide](https://github.com/aws/aws-cdk/wiki/Security-And-Safety-Dev-Guide)

#### Policy report generation

> When `customizeRoles` is used, the `iam-policy-report.txt` report will contain a list
> of IAM roles and associated permissions that would have been created. This report is
> generated so that it attempts to resolve any references and replace with a more user
> friendly value.

Reviewer: nit: phrasing feels awkward
```suggestion
When `customizeRoles` is used, the `iam-policy-report.txt` report will contain a list
of IAM roles and associated permissions that would have been created. This report is
generated in an attempt to resolve and replace any references with a more user-friendly
value.
```
